app/controllers/email_confirmations_controller.rb

```diff
@@ -69,4 +69,8 @@ def confirm_email
   def email_params
     params.permit(email_confirmation: :email).require(:email_confirmation).require(:email)
   end
+
+  def token_params
+    params.permit(:token).fetch(:token, "")
```

Comment:
Wouldn't `params.require(:token)` be enough?

Reply:
`params.require(:token)` checks the presence of a parameter and not the type (will accept an array). Whereas `permit` in this case will not accept an array of token strings and only one string.

Reply:
Then I think you can use `permit(:token).require(:token)`.

Reply:
True! I made the change 👍